### ISSUE TITLE
Incorrect nn.functional.interpolate behavior for align_corners=False

### ISSUE THREAD

**Issue by quinor** [pytorch/pytorch#38579]
## 🐛 Bug

`nn.functional.interpolate` gives incorrect results in some conditions. While the specific example I'm giving out isn't very serious, it did lead to some serious and hard-to-track bugs in my neural network - I don't know how much it fails there.

## To Reproduce

Run the following pytorch code:
```
a = torch.rand(1, 3, 1, 1, dtype=torch.float32)
a = a.half()
b = torch.nn.functional.interpolate(a, size=(5, 5), mode="bilinear", align_corners=False)
```
## Expected behavior

The `b` tensor should have all same values in each of the channels. Instead, some of the values in the last channel differ.
```
a == tensor([[[[0.0952]],
         [[0.5151]],
         [[0.4216]]]], dtype=torch.float16)
b == tensor([[[[0.0952, 0.0952, 0.0952, 0.0952, 0.0952],
          [0.0952, 0.0952, 0.0952, 0.0952, 0.0952],
          [0.0952, 0.0952, 0.0952, 0.0952, 0.0952],
          [0.0952, 0.0952, 0.0952, 0.0952, 0.0952],
          [0.0952, 0.0952, 0.0952, 0.0952, 0.0952]],
         [[0.5151, 0.5151, 0.5151, 0.5151, 0.5151],
          [0.5151, 0.5151, 0.5151, 0.5151, 0.5151],
          [0.5151, 0.5151, 0.5151, 0.5151, 0.5151],
          [0.5151, 0.5151, 0.5151, 0.5151, 0.5151],
          [0.5151, 0.5151, 0.5151, 0.5151, 0.5151]],
         [[0.4216, 0.4216, 0.4216, 0.4216, 0.4214],
          [0.4216, 0.4216, 0.4216, 0.4216, 0.4214],
          [0.4216, 0.4216, 0.4216, 0.4216, 0.4214],
          [0.4216, 0.4216, 0.4216, 0.4216, 0.4214],
          [0.4214, 0.4214, 0.4214, 0.4214, 0.4214]]]], dtype=torch.float16)
```

## Environment

```
Collecting environment information...
PyTorch version: 1.5.0
Is debug build: No
CUDA used to build PyTorch: 10.2

OS: Arch Linux
GCC version: (Arch Linux 9.3.0-1) 9.3.0
CMake version: version 3.17.2

Python version: 3.8
Is CUDA available: Yes
CUDA runtime version: 10.2.89
GPU models and configuration: GPU 0: GeForce GTX 1080 Ti
Nvidia driver version: 440.82
cuDNN version: /usr/lib/libcudnn.so.7.6.5

Versions of relevant libraries:
[pip3] numpy==1.18.2
[pip3] pytorch-lightning==0.7.5
[pip3] torch==1.5.0
[pip3] torchvision==0.6.0
[conda] Could not collect
```

## Additional context

cc @albanD @mruberry

**Comment by ppwwyyxx** [comment]
Likely just an inevitable loss of precision because of float16. (cc @ailzhang )

**Comment by quinor** [comment]
They could be. I had initially rejected this hyphothesis as the errors are a bit more than what I'm used to with fp16 precision loss.

Also, after eliminating the problematic operations some macroscopic behavior of the network changed significantly, but I guess that must have been a different bug that I mislabeled.

**Comment by albanD** [comment]
If you check the error between these values, by removing the first entry for each channel, you get all 0s. So I will agree with @ppwwyyxx and close this as expected loss of precision due to half.

**Comment by quinor** [comment]
Thank you! I had created a (hopefully) better issue describing the symptoms of the problem that I had found. Just linking it for reference: https://github.com/pytorch/pytorch/issues/38788
